llvm/include/llvm/Target/Target.td

```diff
@@ -777,6 +777,10 @@ class Instruction : InstructionEncoding {
   /// SelectionDAG can.
   bit FastISelShouldIgnore = false;
 
+  /// Should DAGISel ignore this instruction. In cases where lowering may
+  /// be done elsewhere or is unneeded, DAGISel may skip over them.
+  bit DAGISelShouldIgnore = false;
```

Comment:
I think the bit should be on the `Pat` class, not the `Instruction` class?

Do you want to disable DAG import of any pattern that uses a given instruction, or disable import for a Pat (an in/out selection pattern) ? If it's the latter then it's the wrong place for this bit.

The reason there is a FastISel bit I think is because some instructions are probably too slow to emit and FastISel needs to skip over them (?)

Reply:
It probably belongs with Pat, I think this is where I got stuck last I looked at doing this

Reply:
Added the new bit to `Pattern`: `bit ISelShouldIgnore = 0`. It can be set on a pattern to avoid -warn-on-skipped-patterns warnings.